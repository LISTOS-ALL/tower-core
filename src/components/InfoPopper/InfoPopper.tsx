import {
    createStyles,
    Paper,
    Popover,
    Theme,
    WithStyles,
    withStyles,
} from '@material-ui/core';
import { PopoverOrigin } from '@material-ui/core/Popover';
import * as React from 'react';

const styles = (theme: Theme) => createStyles({
    popper: {
        minWidth: 150,
        padding: '10px 0px',
    },
    info: {
        minWidth: 225,
        minHeight: 'auto',
    },
    title: {
        opacity: 0.54,
    },
});

interface StyleProps extends WithStyles<typeof styles> {
    theme: Theme;
}

interface PopperProps {
    anchorEl: HTMLElement | null;
    anchorOrigin: PopoverOrigin | undefined;
    transformOrigin: PopoverOrigin | undefined;
    open: boolean;
    handleClose: () => void;
    data: JSX.Element;
    style?: React.CSSProperties;
    // tslint:disable-next-line:no-any
    action?: (e: any) => void;
    transitionDuration?: number;
}

type Props = StyleProps & PopperProps;

const PopperComponent: React.FunctionComponent<Props> = props => {
    const {
        anchorEl,
        open,
        handleClose,
        data,
        classes,
        anchorOrigin,
        transitionDuration,
        transformOrigin,
        action,
        style,
    } = props;

    const id = open ? 'simple-popper' : undefined;

    return (
        <Popover
            action={action}
            id={id}
            open={open}
            anchorEl={anchorEl}
            onClose={handleClose}
            anchorOrigin={anchorOrigin}
            transformOrigin={transformOrigin}
            transitionDuration={transitionDuration}
            className={classes.popper}
        >
            <Paper style={style} className={classes.info}>{data}</Paper>
        </Popover>
    );
};

export const InfoPopper = withStyles(styles, { withTheme: true })(PopperComponent);
