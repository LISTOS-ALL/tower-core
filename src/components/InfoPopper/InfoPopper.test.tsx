import { shallow } from 'enzyme';
import * as React from 'react';
import { InfoPopper } from './InfoPopper';

const defaultProps = {
    anchorEl: null,
    open: false,
    handleClose: jest.fn(),
    data: <div />,
    anchorOrigin: undefined,
    transformOrigin: undefined,
    style: {
        padding: '3px',
    },
};

describe('InfoPopper test', () => {
    it('should render', () => {
        const wrapper = shallow(<InfoPopper {...defaultProps} />);
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).toBeDefined();
    });
});
