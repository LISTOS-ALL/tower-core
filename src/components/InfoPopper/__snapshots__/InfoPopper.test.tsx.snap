// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InfoPopper test should render 1`] = `ShallowWrapper {}`;
